### ISSUE TITLE
`baselineVersions` in `data.json` is outdated

### ISSUE THREAD

**Issue by mhassan1** [Financial-Times/polyfill-useragent-normaliser#172]
The `baselineVersions` in [data.json](https://github.com/Financial-Times/polyfill-useragent-normaliser/blob/3c0fbc4410f539922d520d7ae8b1632f128a69e3/data.json#L6-L23) have not been updated in 3 years and no longer reflect the ">0.1% of global usage" threshold; for example, IE8 - IE10 are all less than that threshold.

**Comment by JakeChampion** [comment]
I'm very happy to review a pull-request which updates the baseline browser to correctly reflect ">0.1% of global usage"

**Comment by romainmenke** [comment]
">0.1%" is a pretty skewed list at the moment.

IE11 won't die and every other major browser is decreasing the time between updates with auto update having been default for some time. Even Apple is now releasing Safari major versions for older OS's.

Maybe better to align this on a minimum capability (e.g. object getters and setters support)?

```
npx browserslist "> 0.1%"           
and_chr 96
and_ff 94
and_qq 10.4
and_uc 12.12
android 4.4.3-4.4.4
chrome 96
chrome 95
chrome 94
chrome 93
chrome 92
chrome 91
chrome 87
chrome 86
chrome 85
chrome 79
chrome 49
edge 96
edge 95
edge 94
firefox 94
firefox 93
ie 11
ios_saf 15.0-15.1
ios_saf 14.5-14.8
ios_saf 14.0-14.4
ios_saf 13.4-13.7
ios_saf 12.2-12.5
ios_saf 10.3
ios_saf 9.3
op_mini all
opera 81
opera 80
safari 15.1
safari 15
safari 14.1
safari 14
safari 13.1
samsung 15.0
samsung 14.0
samsung 13.0
samsung 4
```

**Comment by JakeChampion** [comment]
We could use >0.05% as the cut-off point instead (I'm not sure that's a good idea...)

I think we should choose a set of browser that we can support and have automated tests for, otherwise the amount of manual testing will become too much of a burden for maintainers of the polyfill.io projects.

> Maybe better to align this on a minimum capability (e.g. object getters and setters support)?

That sounds like a good idea, do you know what versions of browsers that would be?

I imagine that the list we end up with will likely be a combination of the above two options.

**Comment by romainmenke** [comment]
Yes, the minimum versions we can run tests for are important. If it can't be tested in CI it will accidentally break.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#browser_compatibility

https://github.com/mdn/browser-compat-data/blob/385a5c3c82432347836b985613cf672a5c166d06/javascript/builtins/Object.json#L391

```
chrome 5
chrome_android 18
edge 12
firefox 4
firefox_android 4
ie 9
opera 11.6
opera_android 12
safari 5.1
safari_ios 6
samsunginternet_android 1.0
webview_android 1
```

Combined with : https://github.com/Financial-Times/polyfill-library/blob/master/test/polyfills/browsers.toml
and with what browserstack currently supports

```
chrome 14
android 4.4
edge 15
firefox 10
ie 9
opera 12.15
safari 7.1
safari_ios 10
```

This last list should be expanded again from the engines to other derivative browsers.

----

Are there any stats from the polyfill service?
I can imagine that keeping detailed stats there is tricky because of the amount of traffic and privacy concerns.

**Comment by JakeChampion** [comment]
> Are there any stats from the polyfill service?
> I can imagine that keeping detailed stats there is tricky because of the amount of traffic and privacy concerns.

unfortunately we do not have stats for the exact reasons you've mentioned

**Comment by romainmenke** [comment]
Another capability that is interesting to use as a baseline is `Proxy`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy#browser_compatibility

Browsers that fully support `Object.defineProperty` and `Proxy` allow us to fully extend native classes without overriding or re-implementing the base class.

IE 11 doesn't support `Proxy` and it can't be polyfilled so this would be a hard break.
I don't think we can make this jump yet, but it might be the most interesting feature set to have available when writing and maintaining polyfills in the future.

_My personal preference is actually to not move the baseline unless we actually have to.
But this is not based on any real world need. For work stuff we don't support anything any browser version older than 2014 (so no IE11)._

-------

Another factor to consider is what it means to not meet the baseline.

https://polyfill.io/v3/polyfill.js?features=Object.getOwnPropertyDescriptors
https://github.com/Financial-Times/polyfill-library/blob/master/polyfills/Object/getOwnPropertyDescriptors/config.toml#L15

| Browser | UA | polyfilled |
| --- | --- | --- |
| IE 9 | `Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)` | ✅ |
| IE 8 | `Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.0; Trident/4.0)` | 🔴 |
| IE 7 | `Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)` | ✅ |

If we align on any capability and write polyfills with this in mind we will ship polyfills to browsers below the baseline, knowing they can't work there.

This is especially problematic if any polyfill contains new syntax or throws initial execution.

----------

Is there any history in another repo of the current baseline?
The current config is from the initial commit in this repo.
But I am guessing it moved here from somewhere else?

Would be interesting to know how it evolved before.

**Comment by JakeChampion** [comment]
There were no rules around updating the baseline supported versions beforez they used to change whenever Andrew and myself felt they should change.
I tried to come up with a rule the last time I updated but I think following a rule might not be th best decision as likely we would need to break it to retain support for certain browsers. I think dropping support for IE 8 and 9 is a good move, maybe even dropping IE 10.
